flopy/utils/gridintersect.py

```diff
@@ -288,57 +311,67 @@ def intersect(
         return rec
 
     def _set_method_get_gridshapes(self):
-        """internal method, set self._get_gridshapes to the certain method for
-        obtaining gridcells."""
+        """internal method, set self._get_gridshapes to the appropriate method
+        for obtaining grid cell geometries."""
         # Set method for obtaining grid shapes
         if self.mfgrid.grid_type == "structured":
-            self._get_gridshapes = self._rect_grid_to_shape_generator
+            self._get_gridshapes = self._rect_grid_to_geoms_cellids
         elif self.mfgrid.grid_type == "vertex":
-            self._get_gridshapes = self._vtx_grid_to_shape_generator
+            self._get_gridshapes = self._vtx_grid_to_geoms_cellids
         elif self.mfgrid.grid_type == "unstructured":
             raise NotImplementedError()
 
-    def _rect_grid_to_shape_generator(self):
-        """internal method, generator yielding shapely polygons for structured
+    def _rect_grid_to_geoms_cellids(self):
+        """internal method, return shapely polygons and cellids for structured
         grid cells.
 
         Returns
         -------
-        generator :
-            generator of shapely Polygons
+        geoms : list
+            list of shapely Polygons
+        cellids : list
+            list of cellids
         """
         shapely_geo = import_optional_dependency("shapely.geometry")
 
+        geoms = []
+        cellids = []
         for i in range(self.mfgrid.nrow):
             for j in range(self.mfgrid.ncol):
                 xy = self.mfgrid.get_cell_vertices(i, j)
-                p = shapely_geo.Polygon(xy)
-                p.name = (i, j)
-                yield p
+                geoms.append(shapely_geo.Polygon(xy))
+                cellids.append(i * self.mfgrid.ncol + j)
```

Comment:
this could be changed to 
```python
cellids.append(self.mfgrid.get_node([0, i, j])[0])
```
to avoid duplicate routines

Reply:
done.